clients/cache/inmemory.go

```diff
@@ -0,0 +1,97 @@
+package cache
+
+import (
+	"context"
+	"sync"
+	"time"
+)
+
+// InMemoryCache is an in-memory implementation of the Cache interface.
+type InMemoryCache struct {
+	data  map[string]cacheItem
+	mutex sync.RWMutex
+}
+
+// Ensure InMemoryCache implements the Cache interface.
+var _ Cache = (*InMemoryCache)(nil)
+
+// cacheItem represents an item stored in the cache.
+type cacheItem struct {
+	data       []byte
+	expiration time.Time
+}
+
+// NewInMemoryCache creates a new instance of InMemoryCache.
+func NewInMemoryCache() *InMemoryCache {
+	return &InMemoryCache{
+		data: make(map[string]cacheItem),
+	}
+}
+
+// Set sets the value of a key in the cache.
+func (c *InMemoryCache) Set(
+	ctx context.Context,
+	key string,
+	data []byte,
+	expiration time.Duration,
+) error {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	expiry := time.Now().Add(expiration)
+
+	if expiration == 0 {
+		// 100 years in the future to prevent expiry
+		expiry = time.Now().AddDate(100, 0, 0)
+	}
+
+	c.data[key] = cacheItem{
+		data:       data,
+		expiration: expiry,
+	}
+
+	return nil
+}
+
+// Get retrieves the value of a key from the cache.
+func (c *InMemoryCache) Get(ctx context.Context, key string) ([]byte, error) {
+	c.mutex.RLock()
```

Comment:
why do we need a mutex to read a value from the cahce?

Reply:
it's in-memory cache implementation it's only used in tests

so it can be helpful if we run tests in parallel or smth like this

@galxy25 so basically it's up to you, do you suggest to remove mutex?